openpype/plugins/publish/integrate.py

```diff
@@ -0,0 +1,835 @@
+                "look",
+                "audio",
+                "yetiRig",
+                "yeticache",
+                "nukenodes",
+                "gizmo",
+                "source",
+                "matchmove",
+                "image",
+                "assembly",
+                "fbx",
+                "textures",
+                "action",
+                "harmony.template",
+                "harmony.palette",
+                "editorial",
+                "background",
+                "camerarig",
+                "redshiftproxy",
+                "effect",
+                "xgen",
+                "hda",
+                "usd",
+                "staticMesh",
+                "skeletalMesh",
+                "mvLook",
+                "mvUsd",
+                "mvUsdComposition",
+                "mvUsdOverride",
+                "simpleUnrealTexture"
+                ]
+    exclude_families = ["clip", "render.farm"]
+    default_template_name = "publish"
+
+    # Representation context keys that should always be written to
+    # the database even if not used by the destination template
+    db_representation_context_keys = [
+        "project", "asset", "task", "subset", "version", "representation",
+        "family", "hierarchy", "username"
+    ]
+
+    # Attributes set by settings
+    template_name_profiles = None
+
+    def process(self, instance):
+
+        # Exclude instances that also contain families from exclude families
+        families = set(get_instance_families(instance))
+        exclude = families & set(self.exclude_families)
+        if exclude:
+            self.log.debug("Instance not integrated due to exclude "
+                           "families found: {}".format(", ".join(exclude)))
+            return
+
+        file_transactions = FileTransaction(log=self.log)
+        try:
+            self.register(instance, file_transactions)
+        except Exception:
+            # clean destination
+            # todo: preferably we'd also rollback *any* changes to the database
+            file_transactions.rollback()
+            self.log.critical("Error when registering", exc_info=True)
+            six.reraise(*sys.exc_info())
+
+        # Finalizing can't rollback safely so no use for moving it to
+        # the try, except.
+        file_transactions.finalize()
+
+    def register(self, instance, file_transactions):
+
+        instance_stagingdir = instance.data.get("stagingDir")
+        if not instance_stagingdir:
+            self.log.info((
+                "{0} is missing reference to staging directory."
+                " Will try to get it from representation."
+            ).format(instance))
+
+        else:
+            self.log.debug(
+                "Establishing staging directory "
+                "@ {0}".format(instance_stagingdir)
+            )
+
+        # Ensure at least one representation is set up for registering.
+        repres = instance.data.get("representations")
```

Comment:
Check for representations and their filtering should be first part. Missing representations should just skip the instance instead of raise an error.